plugin-base/src/main/kotlin/io/spine/tools/gradle/project/ProjectExts.kt

```diff
@@ -122,3 +128,72 @@ public fun Project.configuration(name: String): Configuration = configurations.g
 /** Obtains a configuration by its name. */
 public fun Project.configuration(name: ConfigurationName): Configuration =
     configuration(name.value())
+
+/**
+ * Tells if this project can deal with Java code.
+ *
+ * @return `true` if `java` plugin is installed, `false` otherwise.
+ */
+public fun Project.hasJava(): Boolean = hasCompileTask(Java)
+
+/**
+ * Tells if this project can deal with Kotlin code.
+ *
+ * @return `true` if any of the tasks starts with `"compile"` and ends with `"Kotlin"`.
+ */
+public fun Project.hasKotlin(): Boolean = hasCompileTask(Kotlin)
+
+/**
+ * Tells if this project has a compile task for the given language.
+ */
+public fun Project.hasCompileTask(language: Language): Boolean {
+    val currentTasks = ImmutableList.copyOf(tasks)
+    val compileTask = currentTasks.find {
+        it.name.startsWith("compile") && it.name.endsWith(language.name)
+    }
+    return compileTask != null
+}
+
+/**
+ * Verifies if the project can deal with Java or Kotlin code.
+ *
+ * The current Protobuf support of Kotlin is based on Java codegen.
+ * Therefore, it is likely that Java would be enabled in the project for
+ * Kotlin proto code to be generated.
+ * Though, it may change someday, and Kotlin support for Protobuf would be
+ * self-sufficient. This method assumes such a case when it checks the presence of
+ * Kotlin compilation tasks.
+ *
+ * @see [hasJava]
+ * @see [hasKotlin]
+ */
+public fun Project.hasJavaOrKotlin(): Boolean {
+    if (hasJava()) {
+        return true
+    }
+    return hasKotlin()
+}
+
+/**
+ * Attempts to obtain the Java compilation Gradle task for the given source set.
+ *
+ * Typically, the task is named by a pattern: `compile<SourceSet name>Java`, or just `compileJava`
+ * if the source set name is `"main"`. If the task does not fit this described pattern, this method
+ * will not find it.
+ */
+public fun Project.javaCompileFor(sourceSet: SourceSet): JavaCompile? {
```

Comment:
```suggestion
public fun Project.findJavaCompileFor(sourceSet: SourceSet): JavaCompile? {
```

Why not use `find*` or similar prefix when returning a nullable type?

And the same suggestion to the method below.

Reply:
Makes sense. I'll rename.